src/physics/em/detail/BhabhaEnergyDistribution.i.hh

```diff
@@ -0,0 +1,85 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file BhabhaEnergyDistribution.i.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Algorithms.hh"
+#include "random/distributions/UniformRealDistribution.hh"
+#include "random/distributions/BernoulliDistribution.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with data from MollerBhabhaInteractor.
+ */
+CELER_FUNCTION
+BhabhaEnergyDistribution::BhabhaEnergyDistribution(
+    const MollerBhabhaPointers& shared, const real_type inc_energy)
+    : electron_mass_c_sq_(shared.electron_mass_c_sq)
+    , inc_energy_(inc_energy)
+    , total_energy_(inc_energy + shared.electron_mass_c_sq)
+    , max_energy_fraction_(1.0)
```

Comment:
Since this is a constant, I would just declare your variable as a constexpr member in the class, so you don't have to initialize it here, and as a hint to the compiler:
```
constexpr real_type max_energy_fraction_ = 1.0;
```

Reply:
Not sure if I can make that one `static` as well, so maybe `const`? If we add `constexpr` we define that value at compile time, but for a class we would only know it at instantiation time.

Reply:
OK, so maybe
```c++
static CELER_CONSTEXPR_FUNCTION real_type max_energy_fraction() { return 1; }
```
for Bhabha and 
```c++
static CELER_CONSTEXPR_FUNCTION real_type max_energy_fraction() { return 0.5; }
```
for Moller?